### ISSUE TITLE
PHP Deprecated an PHP Warning

### ISSUE THREAD

**Issue by Marin-Matosevic** [marinsagovac/woocommerce-tcom-payway#31]
Hi,

I'm running the latest version of your plugin (1.8.1) on a client website, alongside the latest versions of WordPress and WooCommerce, and the site is running on PHP 8.2.

I’m seeing a lot of log messages in the debug.log file—every page reload generates about a dozen entries, causing the log file to grow significantly over the course of the day. Is there anything I can adjust on my end to minimize these messages (apart from turning off the log file)? :) 

Thanks!

```
PHP Deprecated:  Creation of dynamic property WC_TPAYWAY::$domain is deprecated in /plugins/woocommerce-tcom-payway-master/classes/class-wc-tpayway.php on line 16
PHP Deprecated:  Creation of dynamic property WC_TPAYWAY::$curlExtension is deprecated in /plugins/woocommerce-tcom-payway-master/classes/class-wc-tpayway.php on line 21
PHP Deprecated:  Creation of dynamic property WC_TPAYWAY::$ratehrkfixed is deprecated in /plugins/woocommerce-tcom-payway-master/classes/class-wc-tpayway.php on line 23
PHP Deprecated:  Creation of dynamic property WC_TPAYWAY::$tecajnaHnbApi is deprecated in /plugins/woocommerce-tcom-payway-master/classes/class-wc-tpayway.php on line 24
PHP Deprecated:  Creation of dynamic property WC_TPAYWAY::$api_version is deprecated in /plugins/woocommerce-tcom-payway-master/classes/class-wc-tpayway.php on line 26
PHP Deprecated:  Creation of dynamic property WC_TPAYWAY::$shop_id is deprecated in /plugins/woocommerce-tcom-payway-master/classes/class-wc-tpayway.php on line 34
PHP Deprecated:  Creation of dynamic property WC_TPAYWAY::$acq_id is deprecated in /plugins/woocommerce-tcom-payway-master/classes/class-wc-tpayway.php on line 35
PHP Deprecated:  Creation of dynamic property WC_TPAYWAY::$pg_domain is deprecated in /plugins/woocommerce-tcom-payway-master/classes/class-wc-tpayway.php on line 36
PHP Deprecated:  Creation of dynamic property WC_TPAYWAY::$checkout_msg is deprecated in /plugins/woocommerce-tcom-payway-master/classes/class-wc-tpayway.php on line 37
PHP Deprecated:  Creation of dynamic property WC_TPAYWAY::$msg is deprecated in /plugins/woocommerce-tcom-payway-master/classes/class-wc-tpayway.php on line 40

PHP Warning:  Undefined array key "Amount" in /plugins/woocommerce-tcom-payway-master/classes/class-wc-tpayway.php on line 496
PHP Warning:  Undefined array key "ShoppingCartID" in /plugins/woocommerce-tcom-payway-master/classes/class-wc-tpayway.php on line 502
```

**Comment by 9jaDevo** [comment]
Can I take up the task? #31 I can fix that. @Marin-Matosevic

**Comment by jovialcore** [comment]
@Marin-Matosevic  I made a [pull request ](https://github.com/marinsagovac/woocommerce-tcom-payway/pull/32) for it... It should be reviewed. If reviewed and approved, it should solve this issue.

**Comment by marinsagovac** [comment]
Done under 1.8.3 version.
Please check.

Can I close?

- [ ] Yes
- [x] No

**Comment by Marin-Matosevic** [comment]
@marinsagovac Oh my, oh my, it crushed the site. :)
The error says:
PHP Fatal error: Access level to WC_TPAYWAY::$title must be public (as in class WC_Payment_Gateway) in /../plugins/woocommerce-tcom-payway-1.8.3/classes/class-wc-tpayway.php.

**Comment by jovialcore** [comment]
sorry about that. on it

**Comment by jovialcore** [comment]
https://github.com/marinsagovac/woocommerce-tcom-payway/pull/36

Made a pull request to that effect.

**Comment by marinsagovac** [comment]
New fix published under 1.8.4 #36 

If is resolved I can close issue.

**Comment by marinsagovac** [comment]
Closed, refer to #34
